lib/vagrant/scp/commands/scp.rb

```diff
@@ -16,7 +16,11 @@ def execute
           with_target_vms(host) do |machine|
             @ssh_info = machine.ssh_info
             raise Vagrant::Errors::SSHNotReady if @ssh_info.nil?
-            user_at_host = "#{@ssh_info[:username]}@#{@ssh_info[:host]}"
+            if @ssh_info[:host].include? ":"
```

Comment:
Please do proper IPv6 check here, because what you are actually checking is just if ":" is present, not if the argument is a valid IPv6 address. Some regexp or something like that would solve the problem.

Reply:
See new PR